### DISCUSSION TITLE
Boussinesq convergence problem

### DISCUSSION THREAD

**Discussion by erikgus94** [idaholab/moose#19208] [Q&A Modules: Navier-Stokes] [answered]
Hi,

I was hoping to get some tips or answers to how I could get a converging solution to my Boussinesq problem. My input file is pretty much identical to the example file on the INSFVMomentumBoussinesq page except that I have redefined the boundary conditions and material properties as well as included a mesh file with a 2d loop. I have tried using different scaling parameters with no luck, my input and mesh files are available [here](https://github.com/erikgus94/Boussinesq-question).

**Comment by GiudGiud** [comment, marked as answer]
hello

I've been able to make a reduced version of this simulation converge on a quad simplified mesh using a transient and a viscosity rampdown through time.
The problem is that the initialization is too difficult for the numerical solver with the real value of viscosity. So we start with a highly viscous problem, basically diffusion with a little bit of flow, and go down from there.

I also added an initial condition to temperature (a little rough going from 0K to 660 over 5s otherwise)
And switched the advected quantity (momentum, energy) interpolation method from average to upwind. Average can be oscillatory as it s a second order method. We will have 2nd order limited (not oscillating) interpolation methods in the near future. 

Finally, I switched from no slip to free slip BCs. The mesh you are using is much too coarse to resolve a no-slip wall.

I pushed my solution to this repo: https://github.com/GiudGiud/Boussinesq-question.

It's likely not converged mesh-wise. The relaxation transient is also not completely finished, but at least it's running with the right fluid properties, we just need to wait.

Note that you will want to optimize the time stepping, the mesh and the numerical solver to get better performance. 
Triangle meshes are supported but we recommend using quad meshes for fluid simulations for now.
I think the bigger issue is your mesh. Switching the mesh from mine to yours breaks down at 10x the viscosity

Guillaume

**Reply by erikgus94** [reply]
Thanks a lot for the help, it is much appreciated! 

Smart solution to use transient rampdown of the viscosity. I'm managing to get an converging solution with the mesh you made @GiudGiud.

I made another mesh file with quad mesh as you suggested but I'm still not able to get it to converge even for the first time step. The mesh file can be found [here](https://github.com/erikgus94/Boussinesq-question/blob/2dd2df7010e6bba2068032cbc87e68cdbb013de2/2d_inner_fluid_quad_corse.e), the new mesh is still coarse but I'm applying free slip BSs. See input file [here](https://github.com/erikgus94/Boussinesq-question/blob/ea795c4e4902ddace777ab351ade4da05cc8e70e/boussinesq32.i).

Could it be that it is actually converging but very slowly?

And also, have I assumed right that the units Navier Stokes module uses  is [Pa* s] for viscosity, [kg/m^3] for density, [W/m* K] for thermal conductivity and [J/kg* K] for specific heat capacity?

**Reply by GiudGiud** [reply]
Yup we use SI units in the Navier Stokes module. Some other MOOSE modules allow for imperial units but it hasnt been done for that one
My guess is that we have to keep helping the solver by initializing better, or using an even slower viscosity rampdown.

Could you please try to get convergence on a non-slanted version of your mesh (so starting from my mesh and adjusting the pipe size until it's the same as your mesh. This will require adjusting all the mesh generators btw) ?
